### PR TITLE
Set the value of prune_at_shallow_depth_on_pv_node on a UCI option callback

### LINKED ISSUE #119: Options["Training"] is read before it can be assigned through uci if Threads option precedes Training
Found by linox. I'm posting this verbatim from discord.

There is a bug in the use of option Options["Training"]. Default value is false, when user sends the command
setoption name training value true it will not set to true because it is called under Search::init() at https://github.com/nodchip/Stockfish/blob/4cc98d80f8a5e6eb0b716a47bc4eb8b877b5a979/src/search.cpp#L199
To fix this issue the training = Options["Training"]; can be called under void MainThread::search() at https://github.com/nodchip/Stockfish/blob/4cc98d80f8a5e6eb0b716a47bc4eb8b877b5a979/src/search.cpp#L219

### PR DESCRIPTION
**Pull request by nodchip**
This pull request contains two changes:
- Changes the option name to prune at shallow depth on PV nodes more descriptive.
   - Training -> PruneAtShallowDepthOnPvNode
- Sets the value of prune_at_shallow_depth_on_pv_node on a UCI option callback.
- Invert true/false of prune_at_shallow_depth_on_pv_node according to the option name.  The default behavior is not changed.

fixes #119

### CODE REVIEW

File: src/search.cpp [hunk 1]
```diff
@@ -1011,7 +1011,9 @@ namespace {
 
       // Step 12. Pruning at shallow depth (~200 Elo)
       if (  !rootNode
-          && !(training && PvNode)
+#ifdef EVAL_LEARN
+          && !(!prune_at_shallow_depth_on_pv_node && PvNode)
```
Comment (noobpwnftw) [inline src/search.cpp:1015]: `&& (PvNode ? prune_at_shallow_depth_on_pv_node : true)`

Otherwise it breaks bench.

**Comment by Sopel97** [comment]
Looks fine, just change the condition as noob pointed out.

File: src/search.cpp (hunk 1 above)
Comment (nodchip) [reply src/search.cpp]: Thank you for the suggestion.  The suggested code looks more intuitive.  I followed it.

**Comment by nodchip** [comment]
Thank you for reviewing.  I fix it.

**Pull request merged**